In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
mergedDescs=pd.read_csv("merged_Descs_sorted.csv")
data=mergedDescs.copy()

In [3]:
test_names=pd.read_csv("ExternalID.csv", header=None)
train_names=pd.read_csv("TrainID.csv", header=None)

In [4]:
data["BCF"] = data.apply(lambda row: 0 if row["logBCF"] < 3.0 else 1, axis=1)

In [5]:
data = data.drop(['logBCF'], axis=1)

In [6]:
data.groupby(['BCF'])['BCF'].count()

BCF
0    798
1    209
Name: BCF, dtype: int64

In [7]:
# for splitting the dataset, selected rows based on unique values in the 'Name' column using or (|)
# unique values were chosen from original datasets provided by authors of publication
test= data[(data['Name']==729)|
           (data['Name']==367)|
           (data['Name']==475)|
           (data['Name']==951)|
           (data['Name']==21)|
           (data['Name']==810)|
           (data['Name']==1009)|
           (data['Name']==273)|
           (data['Name']==168)|
           (data['Name']==586)|
           (data['Name']==594)|
           (data['Name']==22)|
           (data['Name']==13)|
           (data['Name']==941)|
           (data['Name']==4)|
           (data['Name']==253)|
           (data['Name']==748)|
           (data['Name']==807)|
           (data['Name']==23)|
           (data['Name']==953)|
           (data['Name']==876)|
           (data['Name']==571)|
           (data['Name']==600)|
           (data['Name']==372)|
           (data['Name']==492)|
           (data['Name']==802)|
           (data['Name']==3)|
           (data['Name']==118)|
           (data['Name']==606)|
           (data['Name']==42)|
           (data['Name']==569)|
           (data['Name']==505)|
           (data['Name']==324)|
           (data['Name']==588)|
           (data['Name']==173)|
           (data['Name']==65)|
           (data['Name']==867)|
           (data['Name']==145)|
           (data['Name']==1032)|
           (data['Name']==98)|
           (data['Name']==101)|
           (data['Name']==478)|
           (data['Name']==30)|
           (data['Name']==959)|
           (data['Name']==960)|
           (data['Name']==396)|
           (data['Name']==621)|
           (data['Name']==860)|
           (data['Name']==559)|
           (data['Name']==100)|
           (data['Name']==568)|
           (data['Name']==847)|
           (data['Name']==194)|
           (data['Name']==120)|
           (data['Name']==24)|
           (data['Name']==970)|
           (data['Name']==149)|
           (data['Name']==99)|
           (data['Name']==204)|
           (data['Name']==183)|
           (data['Name']==10)|
           (data['Name']==652)|
           (data['Name']==840)|
           (data['Name']==774)|
           (data['Name']==576)|
           (data['Name']==496)|
           (data['Name']==74)|
           (data['Name']==557)|
           (data['Name']==871)|
           (data['Name']==1030)|
           (data['Name']==518)|
           (data['Name']==126)|
           (data['Name']==964)|
           (data['Name']==215)|
           (data['Name']==73)|
           (data['Name']==555)|
           (data['Name']==412)|
           (data['Name']==150)|
           (data['Name']==719)|
           (data['Name']==537)|
           (data['Name']==612)|
           (data['Name']==136)|
           (data['Name']==75)|
           (data['Name']==26)|
           (data['Name']==707)|
           (data['Name']==424)|
           (data['Name']==917)|
           (data['Name']==565)|
           (data['Name']==79)|
           (data['Name']==66)|
           (data['Name']==248)|
           (data['Name']==155)|
           (data['Name']==237)|
           (data['Name']==721)|
           (data['Name']==212)|
           (data['Name']==523)|
           (data['Name']==541)|
           (data['Name']==112)|
           (data['Name']==668)|
           (data['Name']==222)|
           (data['Name']==940)|
           (data['Name']==491)|
           (data['Name']==924)|
           (data['Name']==146)|
           (data['Name']==64)|
           (data['Name']==539)|
           (data['Name']==190)|
           (data['Name']==1007)|
           (data['Name']==511)|
           (data['Name']==854)|
           (data['Name']==334)|
           (data['Name']==617)|
           (data['Name']==409)|
           (data['Name']==582)|
           (data['Name']==61)|
           (data['Name']==156)|
           (data['Name']==1036)|
           (data['Name']==543)|
           (data['Name']==706)|
           (data['Name']==95)|
           (data['Name']==442)|
           (data['Name']==649)|
           (data['Name']==989)|
           (data['Name']==121)|
           (data['Name']==388)|
           (data['Name']==708)|
           (data['Name']==616)|
           (data['Name']==436)|
           (data['Name']==444)|
           (data['Name']==968)|
           (data['Name']==734)|
           (data['Name']==205)|
           (data['Name']==611)|
           (data['Name']==610)|
           (data['Name']==92)|
           (data['Name']==819)|
           (data['Name']==535)|
           (data['Name']==653)|
           (data['Name']==1033)|
           (data['Name']==943)|
           (data['Name']==656)|
           (data['Name']==288)|
           (data['Name']==347)|
           (data['Name']==698)|
           (data['Name']==227)|
           (data['Name']==337)|
           (data['Name']==745)|
           (data['Name']==293)|
           (data['Name']==452)|
           (data['Name']==609)|
           (data['Name']==384)|
           (data['Name']==763)|
           (data['Name']==363)|
           (data['Name']==368)|
           (data['Name']==1006)|
           (data['Name']==546)|
           (data['Name']==137)|
           (data['Name']==797)|
           (data['Name']==762)|
           (data['Name']==1018)|
           (data['Name']==153)|
           (data['Name']==769)|
           (data['Name']==312)|
           (data['Name']==1020)|
           (data['Name']==216)|
           (data['Name']==151)|
           (data['Name']==815)|
           (data['Name']==800)|
           (data['Name']==290)|
           (data['Name']==345)|
           (data['Name']==371)|
           (data['Name']==401)|
           (data['Name']==733)|
           (data['Name']==768)|
           (data['Name']==40)|
           (data['Name']==58)|
           (data['Name']==948)|
           (data['Name']==132)|
           (data['Name']==728)|
           (data['Name']==676)|
           (data['Name']==855)|
           (data['Name']==325)|
           (data['Name']==737)|
           (data['Name']==309)|
           (data['Name']==355)|
           (data['Name']==925)|
           (data['Name']==8)|
           (data['Name']==743)|
           (data['Name']==913)|
           (data['Name']==782)|
           (data['Name']==428)|
           (data['Name']==301)|
           (data['Name']==431)|
           (data['Name']==592)|
           (data['Name']==15)|
           (data['Name']==903)|
           (data['Name']==276)|
           (data['Name']==315)|
           (data['Name']==302)|
           (data['Name']==525)|
           (data['Name']==316)|
           (data['Name']==35)|
           (data['Name']==422)|
           (data['Name']==723)|
           (data['Name']==59)|
           (data['Name']==1021)|
           (data['Name']==184)|
           (data['Name']==53)|
           (data['Name']==892)|
           (data['Name']==69)|
           (data['Name']==973)|
           (data['Name']==322)|
           (data['Name']==310)|
           (data['Name']==623)|
           (data['Name']==965)|
           (data['Name']==124)|
           (data['Name']==420)|
           (data['Name']==16)|
           (data['Name']==882)|
           (data['Name']==961)|
           (data['Name']==90)|
           (data['Name']==283)|
           (data['Name']==839)|
           (data['Name']==39)|
           (data['Name']==856)|
           (data['Name']==752)|
           (data['Name']==1035)|
           (data['Name']==540)|
           (data['Name']==1004)|
           (data['Name']==545)|
           (data['Name']==863)|
           (data['Name']==296)|
           (data['Name']==447)|
           (data['Name']==558)|
           (data['Name']==286)|
           (data['Name']==348)|
           (data['Name']==111)|
           (data['Name']==831)|
           (data['Name']==130)|
           (data['Name']==710)|
           (data['Name']==56)|
           (data['Name']==242)|
           (data['Name']==779)|
           (data['Name']==643)|
           (data['Name']==574)|
           (data['Name']==672)|
           (data['Name']==356)|
           (data['Name']==2)|
           (data['Name']==147)|
           (data['Name']==317)|
           (data['Name']==502)|
           (data['Name']==853)|
           (data['Name']==932)|
           (data['Name']==937)|
           (data['Name']==437)|
           (data['Name']==143)|
           (data['Name']==922)|
           (data['Name']==773)|
           (data['Name']==659)|
           (data['Name']==378)|
           (data['Name']==230)|
           (data['Name']==241)|
           (data['Name']==603)|
           (data['Name']==343)|
           (data['Name']==524)|
           (data['Name']==28)|
           (data['Name']==307)|
           (data['Name']==259)|
           (data['Name']==342)|
           (data['Name']==454)|
           (data['Name']==791)|
           (data['Name']==81)|
           (data['Name']==697)|
           (data['Name']==209)|
           (data['Name']==415)|
           (data['Name']==125)|
           (data['Name']==599)|
           (data['Name']==662)|
           (data['Name']==593)|
           (data['Name']==313)|
           (data['Name']==628)|
           (data['Name']==664)|
           (data['Name']==705)|
           (data['Name']==117)|
           (data['Name']==760)|
           (data['Name']==878)|
           (data['Name']==534)|
           (data['Name']==232)|
           (data['Name']==234)|
           (data['Name']==977)|
           (data['Name']==318)|
           (data['Name']==688)|
           (data['Name']==104)|
           (data['Name']==969)|
           (data['Name']==292)|
           (data['Name']==445)|
           (data['Name']==482)|
           (data['Name']==678)|
           (data['Name']==618)|
           (data['Name']==266)|
           (data['Name']==642)|
           (data['Name']==140)|
           (data['Name']==249)|
           (data['Name']==873)|
           (data['Name']==984)|
           (data['Name']==421)|
           (data['Name']==279)|
           (data['Name']==185)|
           (data['Name']==758)|
           (data['Name']==201)|
           (data['Name']==682)|
           (data['Name']==211)|
           (data['Name']==1029)|
           (data['Name']==981)|
           (data['Name']==887)|
           (data['Name']==133)|
           (data['Name']==391)|
           (data['Name']==684)|
           (data['Name']==670)|
           (data['Name']==694)|
           (data['Name']==641)|
           (data['Name']==247)|
           (data['Name']==256)|
           (data['Name']==686)|
           (data['Name']==218)|
           (data['Name']==893)|
           (data['Name']==875)|
           (data['Name']==18)|
           (data['Name']==679)|
           (data['Name']==280)|
           (data['Name']==999)|
           (data['Name']==323)|
           (data['Name']==687)|
           (data['Name']==683)
          ]

In [8]:
# the same approach as for the test set above
train=data[(data['Name']==485)|
           (data['Name']==381)|
        (data['Name']==282)|
        (data['Name']==333)|
        (data['Name']==275)|
        (data['Name']==494)|
        (data['Name']==161)|
        (data['Name']==584)|
        (data['Name']==823)|
        (data['Name']==410)|
        (data['Name']==352)|
        (data['Name']==908)|
        (data['Name']==261)|
        (data['Name']==715)|
        (data['Name']==551)|
        (data['Name']==597)|
        (data['Name']==711)|
        (data['Name']==716)|
        (data['Name']==846)|
        (data['Name']==103)|
        (data['Name']==224)|
        (data['Name']==54)|
        (data['Name']==792)|
        (data['Name']==1)|
        (data['Name']==457)|
        (data['Name']==730)|
        (data['Name']==829)|
        (data['Name']==869)|
        (data['Name']==219)|
        (data['Name']==601)|
        (data['Name']==105)|
        (data['Name']==37)|
        (data['Name']==148)|
        (data['Name']==532)|
        (data['Name']==416)|
        (data['Name']==509)|
        (data['Name']==453)|
        (data['Name']==468)|
        (data['Name']==179)|
        (data['Name']==154)|
        (data['Name']==615)|
        (data['Name']==771)|
        (data['Name']==14)|
        (data['Name']==195)|
        (data['Name']==265)|
        (data['Name']==640)|
        (data['Name']==129)|
        (data['Name']==335)|
        (data['Name']==455)|
        (data['Name']==477)|
        (data['Name']==498)|
        (data['Name']==573)|
        (data['Name']==1025)|
        (data['Name']==199)|
        (data['Name']==134)|
        (data['Name']==91)|
        (data['Name']==268)|
        (data['Name']==197)|
        (data['Name']==515)|
        (data['Name']==80)|
        (data['Name']==508)|
        (data['Name']==838)|
        (data['Name']==461)|
        (data['Name']==722)|
        (data['Name']==203)|
        (data['Name']==604)|
        (data['Name']==41)|
        (data['Name']==11)|
        (data['Name']==48)|
        (data['Name']==872)|
        (data['Name']==198)|
        (data['Name']==735)|
        (data['Name']==229)|
        (data['Name']==906)|
        (data['Name']==390)|
        (data['Name']==581)|
        (data['Name']==598)|
        (data['Name']==1031)|
        (data['Name']==221)|
        (data['Name']==107)|
        (data['Name']==57)|
        (data['Name']==258)|
        (data['Name']==171)|
        (data['Name']==693)|
        (data['Name']==645)|
        (data['Name']==31)|
        (data['Name']==884)|
        (data['Name']==210)|
        (data['Name']==563)|
        (data['Name']==650)|
        (data['Name']==361)|
        (data['Name']==364)|
        (data['Name']==828)|
        (data['Name']==12)|
        (data['Name']==564)|
        (data['Name']==174)|
        (data['Name']==536)|
        (data['Name']==97)|
        (data['Name']==519)|
        (data['Name']==483)|
        (data['Name']==963)|
        (data['Name']==277)|
        (data['Name']==583)|
        (data['Name']==552)|
        (data['Name']==514)|
        (data['Name']==119)|
        (data['Name']==304)|
        (data['Name']==362)|
        (data['Name']==213)|
        (data['Name']==71)|
        (data['Name']==231)|
        (data['Name']==51)|
        (data['Name']==942)|
        (data['Name']==1019)|
        (data['Name']==113)|
        (data['Name']==660)|
        (data['Name']==510)|
        (data['Name']==142)|
        (data['Name']==47)|
        (data['Name']==966)|
        (data['Name']==985)|
        (data['Name']==157)|
        (data['Name']==17)|
        (data['Name']==46)|
        (data['Name']==799)|
        (data['Name']==544)|
        (data['Name']==139)|
        (data['Name']==400)|
        (data['Name']==62)|
        (data['Name']==252)|
        (data['Name']==786)|
        (data['Name']==820)|
        (data['Name']==251)|
        (data['Name']==862)|
        (data['Name']==935)|
        (data['Name']==671)|
        (data['Name']==624)|
        (data['Name']==633)|
        (data['Name']==978)|
        (data['Name']==383)|
        (data['Name']==516)|
        (data['Name']==891)|
        (data['Name']==931)|
        (data['Name']==178)|
        (data['Name']==202)|
        (data['Name']==562)|
        (data['Name']==625)|
        (data['Name']==60)|
        (data['Name']==96)|
        (data['Name']==920)|
        (data['Name']==547)|
        (data['Name']==351)|
        (data['Name']==635)|
        (data['Name']==776)|
        (data['Name']==228)|
        (data['Name']==45)|
        (data['Name']==88)|
        (data['Name']==192)|
        (data['Name']==44)|
        (data['Name']==549)|
        (data['Name']==70)|
        (data['Name']==499)|
        (data['Name']==67)|
        (data['Name']==696)|
        (data['Name']==602)|
        (data['Name']==36)|
        (data['Name']==952)|
        (data['Name']==975)|
        (data['Name']==848)|
        (data['Name']==440)|
        (data['Name']==1005)|
        (data['Name']==94)|
        (data['Name']==550)|
        (data['Name']==837)|
        (data['Name']==1015)|
        (data['Name']==507)|
        (data['Name']==780)|
        (data['Name']==34)|
        (data['Name']==527)|
        (data['Name']==407)|
        (data['Name']==858)|
        (data['Name']==577)|
        (data['Name']==956)|
        (data['Name']==667)|
        (data['Name']==1022)|
        (data['Name']==162)|
        (data['Name']==620)|
        (data['Name']==651)|
        (data['Name']==830)|
        (data['Name']==513)|
        (data['Name']==865)|
        (data['Name']==188)|
        (data['Name']==298)|
        (data['Name']==626)|
        (data['Name']==548)|
        (data['Name']==303)|
        (data['Name']==84)|
        (data['Name']==947)|
        (data['Name']==566)|
        (data['Name']==954)|
        (data['Name']==506)|
        (data['Name']==267)|
        (data['Name']==319)|
        (data['Name']==353)|
        (data['Name']==929)|
        (data['Name']==83)|
        (data['Name']==833)|
        (data['Name']==9)|
        (data['Name']==967)|
        (data['Name']==880)|
        (data['Name']==957)|
        (data['Name']==27)|
        (data['Name']==822)|
        (data['Name']==979)|
        (data['Name']==332)|
        (data['Name']==1003)|
        (data['Name']==556)|
        (data['Name']==843)|
        (data['Name']==888)|
        (data['Name']==5)|
        (data['Name']==533)|
        (data['Name']==330)|
        (data['Name']==781)|
        (data['Name']==927)|
        (data['Name']==526)|
        (data['Name']==25)|
        (data['Name']==114)|
        (data['Name']==945)|
        (data['Name']==834)|
        (data['Name']==764)|
        (data['Name']==859)|
        (data['Name']==223)|
        (data['Name']==375)|
        (data['Name']==1010)|
        (data['Name']==832)|
        (data['Name']==425)|
        (data['Name']==634)|
        (data['Name']==704)|
        (data['Name']==844)|
        (data['Name']==87)|
        (data['Name']==877)|
        (data['Name']==20)|
        (data['Name']==852)|
        (data['Name']==1001)|
        (data['Name']==938)|
        (data['Name']==495)|
        (data['Name']==665)|
        (data['Name']==427)|
        (data['Name']==845)|
        (data['Name']==135)|
        (data['Name']==585)|
        (data['Name']==365)|
        (data['Name']==918)|
        (data['Name']==187)|
        (data['Name']==169)|
        (data['Name']==474)|
        (data['Name']==619)|
        (data['Name']==939)|
        (data['Name']==567)|
        (data['Name']==666)|
        (data['Name']==864)|
        (data['Name']==486)|
        (data['Name']==914)|
        (data['Name']==622)|
        (data['Name']==849)|
        (data['Name']==836)|
        (data['Name']==614)|
        (data['Name']==467)|
        (data['Name']==340)|
        (data['Name']==759)|
        (data['Name']==481)|
        (data['Name']==374)|
        (data['Name']==1008)|
        (data['Name']==77)|
        (data['Name']==958)|
        (data['Name']==189)|
        (data['Name']==528)|
        (data['Name']==1012)|
        (data['Name']==226)|
        (data['Name']==466)|
        (data['Name']==1002)|
        (data['Name']==419)|
        (data['Name']==257)|
        (data['Name']==271)|
        (data['Name']==373)|
        (data['Name']==393)|
        (data['Name']==191)|
        (data['Name']==572)|
        (data['Name']==49)|
        (data['Name']==287)|
        (data['Name']==857)|
        (data['Name']==472)|
        (data['Name']==811)|
        (data['Name']==790)|
        (data['Name']==850)|
        (data['Name']==934)|
        (data['Name']==661)|
        (data['Name']==784)|
        (data['Name']==632)|
        (data['Name']==955)|
        (data['Name']==200)|
        (data['Name']==55)|
        (data['Name']==736)|
        (data['Name']==928)|
        (data['Name']==68)|
        (data['Name']==43)|
        (data['Name']==500)|
        (data['Name']==1034)|
        (data['Name']==180)|
        (data['Name']==439)|
        (data['Name']==497)|
        (data['Name']==264)|
        (data['Name']==450)|
        (data['Name']==269)|
        (data['Name']==731)|
        (data['Name']==911)|
        (data['Name']==403)|
        (data['Name']==542)|
        (data['Name']==1013)|
        (data['Name']==673)|
        (data['Name']==115)|
        (data['Name']==561)|
        (data['Name']==122)|
        (data['Name']==331)|
        (data['Name']==578)|
        (data['Name']==405)|
        (data['Name']==714)|
        (data['Name']==812)|
        (data['Name']==742)|
        (data['Name']==423)|
        (data['Name']==756)|
        (data['Name']==321)|
        (data['Name']==746)|
        (data['Name']==208)|
        (data['Name']==738)|
        (data['Name']==278)|
        (data['Name']==285)|
        (data['Name']==306)|
        (data['Name']==327)|
        (data['Name']==350)|
        (data['Name']==357)|
        (data['Name']==379)|
        (data['Name']==399)|
        (data['Name']==433)|
        (data['Name']==448)|
        (data['Name']==744)|
        (data['Name']==750)|
        (data['Name']==821)|
        (data['Name']==1016)|
        (data['Name']==910)|
        (data['Name']==933)|
        (data['Name']==596)|
        (data['Name']==675)|
        (data['Name']==490)|
        (data['Name']==349)|
        (data['Name']==1028)|
        (data['Name']==554)|
        (data['Name']==712)|
        (data['Name']==504)|
        (data['Name']==813)|
        (data['Name']==1024)|
        (data['Name']==976)|
        (data['Name']==72)|
        (data['Name']==426)|
        (data['Name']==108)|
        (data['Name']==255)|
        (data['Name']==326)|
        (data['Name']==339)|
        (data['Name']==449)|
        (data['Name']==469)|
        (data['Name']==971)|
        (data['Name']==1017)|
        (data['Name']==946)|
        (data['Name']==575)|
        (data['Name']==131)|
        (data['Name']==487)|
        (data['Name']==438)|
        (data['Name']==669)|
        (data['Name']==700)|
        (data['Name']==329)|
        (data['Name']==775)|
        (data['Name']==842)|
        (data['Name']==522)|
        (data['Name']==972)|
        (data['Name']==1014)|
        (data['Name']==883)|
        (data['Name']==912)|
        (data['Name']==233)|
        (data['Name']==717)|
        (data['Name']==186)|
        (data['Name']==32)|
        (data['Name']==629)|
        (data['Name']==370)|
        (data['Name']==489)|
        (data['Name']==354)|
        (data['Name']==727)|
        (data['Name']==291)|
        (data['Name']==1026)|
        (data['Name']==464)|
        (data['Name']==236)|
        (data['Name']==369)|
        (data['Name']==754)|
        (data['Name']==789)|
        (data['Name']==962)|
        (data['Name']==785)|
        (data['Name']==926)|
        (data['Name']==553)|
        (data['Name']==359)|
        (data['Name']==206)|
        (data['Name']==827)|
        (data['Name']==909)|
        (data['Name']==320)|
        (data['Name']==560)|
        (data['Name']==116)|
        (data['Name']==709)|
        (data['Name']==414)|
        (data['Name']==644)|
        (data['Name']==605)|
        (data['Name']==907)|
        (data['Name']==806)|
        (data['Name']==591)|
        (data['Name']==479)|
        (data['Name']==608)|
        (data['Name']==788)|
        (data['Name']==50)|
        (data['Name']==757)|
        (data['Name']==93)|
        (data['Name']==109)|
        (data['Name']==377)|
        (data['Name']==127)|
        (data['Name']==366)|
        (data['Name']==861)|
        (data['Name']==681)|
        (data['Name']==801)|
        (data['Name']==851)|
        (data['Name']==930)|
        (data['Name']==803)|
        (data['Name']==76)|
        (data['Name']==870)|
        (data['Name']==63)|
        (data['Name']==33)|
        (data['Name']==777)|
        (data['Name']==986)|
        (data['Name']==1023)|
        (data['Name']==284)|
        (data['Name']==718)|
        (data['Name']==724)|
        (data['Name']==295)|
        (data['Name']==783)|
        (data['Name']==905)|
        (data['Name']==476)|
        (data['Name']==772)|
        (data['Name']==809)|
        (data['Name']==835)|
        (data['Name']==164)|
        (data['Name']==761)|
        (data['Name']==825)|
        (data['Name']==235)|
        (data['Name']==488)|
        (data['Name']==82)|
        (data['Name']==473)|
        (data['Name']==392)|
        (data['Name']==896)|
        (data['Name']==281)|
        (data['Name']==299)|
        (data['Name']==7)|
        (data['Name']==607)|
        (data['Name']==658)|
        (data['Name']==382)|
        (data['Name']==314)|
        (data['Name']==170)|
        (data['Name']==794)|
        (data['Name']==720)|
        (data['Name']==1011)|
        (data['Name']==459)|
        (data['Name']==990)|
        (data['Name']==879)|
        (data['Name']==732)|
        (data['Name']==78)|
        (data['Name']==787)|
        (data['Name']==86)|
        (data['Name']==916)|
        (data['Name']==176)|
        (data['Name']==297)|
        (data['Name']==766)|
        (data['Name']==531)|
        (data['Name']==796)|
        (data['Name']==1027)|
        (data['Name']==982)|
        (data['Name']==866)|
        (data['Name']==336)|
        (data['Name']==165)|
        (data['Name']==740)|
        (data['Name']==239)|
        (data['Name']==646)|
        (data['Name']==85)|
        (data['Name']==755)|
        (data['Name']==451)|
        (data['Name']==220)|
        (data['Name']==713)|
        (data['Name']==418)|
        (data['Name']==983)|
        (data['Name']==380)|
        (data['Name']==590)|
        (data['Name']==432)|
        (data['Name']==570)|
        (data['Name']==463)|
        (data['Name']==595)|
        (data['Name']==529)|
        (data['Name']==770)|
        (data['Name']==395)|
        (data['Name']==895)|
        (data['Name']==214)|
        (data['Name']==89)|
        (data['Name']==793)|
        (data['Name']==739)|
        (data['Name']==741)|
        (data['Name']==753)|
        (data['Name']==460)|
        (data['Name']==874)|
        (data['Name']==484)|
        (data['Name']==358)|
        (data['Name']==29)|
        (data['Name']==638)|
        (data['Name']==346)|
        (data['Name']==881)|
        (data['Name']==376)|
        (data['Name']==446)|
        (data['Name']==38)|
        (data['Name']==995)|
        (data['Name']==949)|
        (data['Name']==919)|
        (data['Name']==167)|
        (data['Name']==805)|
        (data['Name']==980)|
        (data['Name']==458)|
        (data['Name']==580)|
        (data['Name']==394)|
        (data['Name']==443)|
        (data['Name']==398)|
        (data['Name']==886)|
        (data['Name']==996)|
        (data['Name']==465)|
        (data['Name']==244)|
        (data['Name']==703)|
        (data['Name']==841)|
        (data['Name']==138)|
        (data['Name']==207)|
        (data['Name']==19)|
        (data['Name']==480)|
        (data['Name']==411)|
        (data['Name']==102)|
        (data['Name']==663)|
        (data['Name']==904)|
        (data['Name']==767)|
        (data['Name']==579)|
        (data['Name']==974)|
        (data['Name']==587)|
        (data['Name']==344)|
        (data['Name']==123)|
        (data['Name']==441)|
        (data['Name']==402)|
        (data['Name']==814)|
        (data['Name']==462)|
        (data['Name']==921)|
        (data['Name']==385)|
        (data['Name']==778)|
        (data['Name']==950)|
        (data['Name']==655)|
        (data['Name']==701)|
        (data['Name']==795)|
        (data['Name']==816)|
        (data['Name']==181)|
        (data['Name']==901)|
        (data['Name']==936)|
        (data['Name']==456)|
        (data['Name']==885)|
        (data['Name']==196)|
        (data['Name']==294)|
        (data['Name']==141)|
        (data['Name']==408)|
        (data['Name']==900)|
        (data['Name']==630)|
        (data['Name']==1000)|
        (data['Name']==387)|
        (data['Name']==152)|
        (data['Name']==413)|
        (data['Name']==915)|
        (data['Name']==699)|
        (data['Name']==890)|
        (data['Name']==818)|
        (data['Name']==128)|
        (data['Name']==328)|
        (data['Name']==270)|
        (data['Name']==308)|
        (data['Name']==749)|
        (data['Name']==804)|
        (data['Name']==631)|
        (data['Name']==674)|
        (data['Name']==889)|
        (data['Name']==589)|
        (data['Name']==144)|
        (data['Name']==225)|
        (data['Name']==158)|
        (data['Name']==289)|
        (data['Name']==654)|
        (data['Name']==826)|
        (data['Name']==272)|
        (data['Name']==52)|
        (data['Name']==637)|
        (data['Name']==691)|
        (data['Name']==689)|
        (data['Name']==520)|
        (data['Name']==657)|
        (data['Name']==471)|
        (data['Name']==166)|
        (data['Name']==300)|
        (data['Name']==765)|
        (data['Name']==987)|
        (data['Name']==639)|
        (data['Name']==944)|
        (data['Name']==994)|
        (data['Name']==274)|
        (data['Name']==254)|
        (data['Name']==159)|
        (data['Name']==240)|
        (data['Name']==106)|
        (data['Name']==530)|
        (data['Name']==868)|
        (data['Name']==429)|
        (data['Name']==260)|
        (data['Name']==417)|
        (data['Name']==389)|
        (data['Name']==690)|
        (data['Name']==193)|
        (data['Name']==685)|
        (data['Name']==538)|
        (data['Name']==406)|
        (data['Name']==217)|
        (data['Name']==648)|
        (data['Name']==680)|
        (data['Name']==798)|
        (data['Name']==898)|
        (data['Name']==677)|
        (data['Name']==360)|
        (data['Name']==163)|
        (data['Name']==110)|
        (data['Name']==386)|
        (data['Name']==899)|
        (data['Name']==991)|
        (data['Name']==923)|
        (data['Name']==692)|
        (data['Name']==246)|
        (data['Name']==647)|
        (data['Name']==992)|
        (data['Name']==172)|
        (data['Name']==250)|
        (data['Name']==894)|
        (data['Name']==434)|
        (data['Name']==182)|
        (data['Name']==897)|
        (data['Name']==993)|
        (data['Name']==245)|
        (data['Name']==751)|
        (data['Name']==243)|
        (data['Name']==997)|
        (data['Name']==998)|
        (data['Name']==988)|
        (data['Name']==430)|
        (data['Name']==435)|
        (data['Name']==747)|
        (data['Name']==305)|
        (data['Name']==512)]

In [9]:
count = 0
test1 = pd.DataFrame()
train1 = pd.DataFrame()
for index, row in train.iterrows():
    count += 1
    if (count%3)==0:
        test1 = test1.append(row)
    else:
        train1 = train1.append(row)

In [10]:
train1 = train1[["TopoPSA", "WPATH", "XLogP3", "BCF"]]
test1 = test1[["TopoPSA", "WPATH", "XLogP3", "BCF"]]

In [11]:
X_train = train1.drop(['BCF'], axis=1)
y_train = train1['BCF']
X_test = test1.drop(['BCF'], axis=1)
y_test = test1['BCF']

# 2. SVM (Kernel)

# 2.2. Kernel - Sigmoid

# 2.2.1. Imbalanced Model towards nB compounds

In [12]:
from sklearn.svm import SVC
svclassifier = SVC(kernel='sigmoid')
svclassifier.fit(X_train, y_train)

C:\Users\PRASOON\Anaconda3\lib\site-packages\sklearn\svm\base.py:193: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='auto_deprecated',
    kernel='sigmoid', max_iter=-1, probability=False, random_state=None,
    shrinking=True, tol=0.001, verbose=False)

In [13]:
from sklearn.metrics import classification_report, confusion_matrix

y_pred = svclassifier.predict(X_test)

print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))

[[177   0]
 [ 47   0]]
              precision    recall  f1-score   support

         0.0       0.79      1.00      0.88       177
         1.0       0.00      0.00      0.00        47

    accuracy                           0.79       224
   macro avg       0.40      0.50      0.44       224
weighted avg       0.62      0.79      0.70       224



C:\Users\PRASOON\Anaconda3\lib\site-packages\sklearn\metrics\classification.py:1437: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)


| ACTUAL\PREDICTED          | non-Bioaccumulative (0) | Bioaccumulative (1) | TOTAL |
|---------------------------|---------------------|-------------------------|-------|
|    non-Bioaccumulative (0)|      177             |        0               |   177   |
|  Bioaccumulative (1)      |      47             |        0              |   47  |
|        TOTAL              |      224             |        0              |  224  |

# 2.2.2. Balanced Model

In [14]:
train1.groupby(['BCF'])['BCF'].count()

BCF
0.0    355
1.0     94
Name: BCF, dtype: int64

In [15]:
positive_cases = train1[train1['BCF'] == 1]
negative_sample = train1[train1['BCF'] == 0].sample(94, random_state=0)
train1_balanced = pd.concat([positive_cases,negative_sample])
train1_balanced.sort_index(inplace=True)

X_train_bal = train1_balanced.drop(['BCF'], axis=1)
y_train_bal = train1_balanced['BCF']

train1_balanced.groupby(['BCF'])['BCF'].count()

BCF
0.0    94
1.0    94
Name: BCF, dtype: int64

In [16]:
from sklearn.svm import SVC
svclassifier = SVC(kernel='sigmoid')
svclassifier.fit(X_train_bal, y_train_bal)

C:\Users\PRASOON\Anaconda3\lib\site-packages\sklearn\svm\base.py:193: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='auto_deprecated',
    kernel='sigmoid', max_iter=-1, probability=False, random_state=None,
    shrinking=True, tol=0.001, verbose=False)

In [17]:
from sklearn.metrics import classification_report, confusion_matrix

y_pred = svclassifier.predict(X_test)

print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))

[[  0 177]
 [  0  47]]
              precision    recall  f1-score   support

         0.0       0.00      0.00      0.00       177
         1.0       0.21      1.00      0.35        47

    accuracy                           0.21       224
   macro avg       0.10      0.50      0.17       224
weighted avg       0.04      0.21      0.07       224



C:\Users\PRASOON\Anaconda3\lib\site-packages\sklearn\metrics\classification.py:1437: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)


| ACTUAL\PREDICTED          | non-Bioaccumulative (0) | Bioaccumulative (1) | TOTAL |
|---------------------------|---------------------|-------------------------|-------|
|    non-Bioaccumulative (0)|      0             |        177               |   177   |
|  Bioaccumulative (1)      |      0             |        47              |   47  |
|        TOTAL              |      0             |        224              |   224  |